gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/PrestoResourceManager.java

```diff
@@ -0,0 +1,82 @@
+package com.lyft.data.gateway.ha.router;
+
+import java.util.List;
+import lombok.Data;
+import lombok.ToString;
+
+public interface PrestoResourceManager {
+  ResourceGroupDetail createResourceGroup(ResourceGroupDetail resourceGroup);
+
+  List<ResourceGroupDetail> readResourceGroup();
+
+  ResourceGroupDetail updateResourceGroup(ResourceGroupDetail resourceGroup);
+
+  void deleteResourceGroup(long resourceGroupId);
+
+  SelectorDetail createSelector(SelectorDetail selector);
+
+  List<SelectorDetail> readSelector();
+
+  SelectorDetail updateSelector(SelectorDetail selector);
+
+  void deleteSelector(long resourceGroupId);
+
+  //  void readGlobalProperty();
+  //  void updateGlobalProperty();
+
+  @Data
+  @ToString
+  class ResourceGroupDetail implements Comparable<ResourceGroupDetail> {
```

Comment:
I noticed in the tests that you are able to create an instance of this object even without an id. 

You need to impose restrictions here using Lombok annotations. Please look up RequiredArgsConstructor annotation and add it here with final added for the required fields. 

https://projectlombok.org/features/constructor

Same comment for detail classes of other tables.